### ISSUE TITLE
Add SWE non-dimensionalization scaling module

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#157]
## Problem

The current PDE residual is computed in dimensional form, which means that changing `physics.g` (e.g. from 1 to 9.81) drastically changes the magnitude of flux and source terms. Since loss weights were HPO-tuned for `g=1`, switching to physical `g=9.81` makes the PDE loss ~100x larger, breaking the loss balance and slowing convergence.

More fundamentally, dimensional PDE residuals produce ill-conditioned loss landscapes where different terms span orders of magnitude, requiring careful per-variable loss weighting.

## Proposed Solution

Implement a **non-dimensionalization module** (`src/physics/scaling.py`) that transforms the conservative-form 2D SWE into a dimensionless system where all terms are O(1).

### Characteristic Scales

| Scale | Symbol | Definition |
|-------|--------|------------|
| Length | L₀ | `max(Lx, Ly)` from domain geometry |
| Depth | H₀ | Reference depth (configurable, default 1.0 m) |
| Velocity | U₀ = √(gH₀) | Shallow-water wave celerity |
| Time | T₀ = L₀ / U₀ | Advective timescale |

### Non-dimensional Variables

- **Inputs**: `x* = x/L₀`, `y* = y/L₀`, `t* = t/T₀`
- **Outputs**: `h* = h/H₀`, `(hu)* = hu/(H₀U₀)`, `(hv)* = hv/(H₀U₀)`
- **Bed**: `z* = z_b/H₀`

### Key Result

By choosing `U₀ = √(gH₀)`, gravity `g` vanishes from the flux terms entirely. The **only dimensionless group** is the friction number:

```
C_f = g · n² · L₀ / H₀^(4/3)
```

The PDE residual is computed entirely in starred variables, so every term is O(1) and the loss landscape is well-conditioned.

## Implementation Plan

1. **New module**: `src/physics/scaling.py` — `SWEScaler` class
   - Computes L₀, H₀, U₀, T₀, C_f from config
   - `scale_inputs(x, y, t)` → `(x*, y*, t*)`
   - `scale_outputs(h, hu, hv)` → `(h*, (hu)*, (hv)*)`
   - `unscale_outputs(h*, (hu)*, (hv)*)` → `(h, hu, hv)` for evaluation/plotting
   - `scale_bed(z_b)` → `z*`
   - `dimensionless_friction` property → C_f
2. **Config**: Add optional `scaling.H0` key (default 1.0)
3. **Integration**: Apply in experiment_1 `train.py` and `train_imp_samp.py`
   - Scale collocation points before feeding to network
   - Scale validation targets for metric computation
   - Unscale predictions for plotting
   - Pass C_f instead of g/n to PDE loss

## Benefits

- Removes g as a sensitive hyperparameter
- Loss terms naturally O(1) without manual weight tuning
- Better-conditioned optimization landscape
- Standard practice in computational fluid dynamics

## Config Example

```yaml
scaling:
  H0: 1.0  # reference depth in metres
```
